app/controllers/application_controller.rb

```diff
@@ -1,3 +1,13 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
+
+  def authenticate_user
+    redirect_to sign_in_path unless current_user
+  end
+
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
```

Comment:
You should cache the result of this method. It is frequently used/called so you want the result to be cached in memory. You can use memoization for that purpose.

```ruby
def current_user
  @current_user ||= User.find(session[:user_id]) if session[:user_id]
end
```